zulipterminal/ui_tools/boxes.py

```diff
@@ -1690,23 +1700,45 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             # on message body editing.
             msg_body_edit_enabled = True
             if self.model.initial_data["realm_message_content_edit_limit_seconds"] > 0:
-                time_since_msg_sent = time() - self.message["timestamp"]
-                edit_time_limit = self.model.initial_data[
-                    "realm_message_content_edit_limit_seconds"
-                ]
-                if time_since_msg_sent >= edit_time_limit:
-                    if self.message["type"] == "private":
-                        self.model.controller.report_error(
-                            " Time Limit for editing the message has been exceeded."
-                        )
-                        return key
-                    elif self.message["type"] == "stream":
+                if self.message["sender_id"] == self.model.user_id:
+                    time_since_msg_sent = time() - self.message["timestamp"]
+                    edit_time_limit = self.model.initial_data[
+                        "realm_message_content_edit_limit_seconds"
+                    ]
+                    # Don't allow editing message body if time-limit exceeded.
+                    if time_since_msg_sent >= edit_time_limit:
+                        if self.message["type"] == "private":
+                            self.model.controller.report_error(
+                                " Time Limit for editing the message has been exceeded."
+                            )
+                            return key
+                        elif self.message["type"] == "stream":
+                            self.model.controller.report_warning(
+                                " Only topic editing allowed."
+                                " Time Limit for editing the message body"
+                                " has been exceeded."
+                            )
+                            msg_body_edit_enabled = False
+                elif self.message["type"] == "stream":
+                    # Allow editing topic if the message has "(no topic)"
+                    if self.message["subject"] == "(no topic)":
                         self.model.controller.report_warning(
-                            " Only topic editing allowed."
-                            " Time Limit for editing the message body has"
-                            " been exceeded.",
+                            " Only topic editing is allowed."
```

Comment:
I'm inclined to think that we should indicate why it's allowed - this is someone else's message but with no-topic?